sourcegraph/sourcegraph
# Wildcard V2: <Popover /> style migration

## Description
Manual Migrations to <Popover /> wildcard component, in highlighted files and components with usage of `dropdown-menu`

## Refs
[Gitstart Ticket](https://app.gitstart.com/clients/sourcegraph/tickets/SG-27695)
[SourceGraph Link](https://github.com/sourcegraph/sourcegraph/issues/27695)
[Review Changes](https://www.chromatic.com/pullrequest?appId=5f0f381c0e50750022dc6bf7&number=30231&view=changes)
[Loom video](https://www.loom.com/share/9cf7574f320d43f1b8794167dbdd2a17)
[Parent Issue](https://github.com/sourcegraph/sourcegraph/issues/29200)

## Test Plan
 - Run web app using preview app.
 - Confirm there are no UI variations while comparing all use cases of Popover with prod environment


## Preview App
[Frontend](https://client-sourcegraph-frontend-pr-239.onrender.com/)
[Storybook](https://client-sourcegraph-storybook-pr-239.onrender.com/)

## Implementation
### Styles migration:
1. Migrate usage of `dropdown-menu` class to Wildcard Popover
1. Remove `dropdown-menu` Bootstrap dependency - reimplement styles in Popover CSS module

Note: `dropdown-menu` usage is also required for the `Menu` components. We should wait until this migration (https://github.com/sourcegraph/sourcegraph/issues/27701) is finished before migrating away from global styles.

## App preview:

- [Web](https://sg-web-contractors-sg-27695-1.onrender.com)
- [Storybook](https://5f0f381c0e50750022dc6bf7-mlyraxcmss.chromatic.com)

Check out the [client app preview documentation](https://docs.sourcegraph.com/dev/how-to/client_pr_previews) to learn more.
